test/unit/math/opencl/kernel_generator/opencl_code_test.cpp

```diff
@@ -0,0 +1,62 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/copy.hpp>
+#include <test/unit/math/opencl/kernel_generator/reference_kernel.hpp>
+#include <stan/math.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <string>
+
+const char opencl_code_plus_minus[] = STRINGIFY(
+    double pm_res1 = pm_in1 + pm_in2; double pm_res2 = pm_in1 - pm_in2;);
+
+TEST(KernelGenerator, opencl_code_test) {
+  using Eigen::MatrixXd;
+  using Eigen::VectorXd;
+  using stan::math::matrix_cl;
+  VectorXd m1(6, 1);
+  m1 << 1.1, 1.2, 1.3, 1.4, 1.5, 1.6;
+  VectorXd m2(6, 1);
+  m2 << 1.5, 1.6, 1.1, 1.2, 1.3, 1.4;
+
+  matrix_cl<double> m1_cl(m1);
+  matrix_cl<double> m2_cl(m2);
+
+  auto code = stan::math::opencl_code<opencl_code_plus_minus>(
+      std::make_tuple("pm_in1", "pm_in2"), m1_cl, m2_cl);
+
+  matrix_cl<double> res1_cl;
+  matrix_cl<double> res2_cl;
+
+  stan::math::results(res1_cl, res2_cl) = stan::math::expressions(
```

Comment:
What happens if a custom kernel  is input two args of uneven length?

Reply:
Huh, I haven't thought about that. For now I think we can require them to have the same size, but I also have some ideas about how to allow different sizes in the future.

Reply:
Ok, cool with that.